### PR TITLE
Fixes and Rev4 updates for PAM controls

### PR DESCRIPTION
**Pull request by samcornwell**
These are all the changes we have made in our repo at https://github.com/djhaynes/inspec-profile-disa_stig-el7 relating to PAM controls. This includes updates of these controls to RHEL7 STIG Rev4, as well as various fixes.

### CODE REVIEW

File: controls/V-71933.rb [hunk 1]
```diff
@@ -1,64 +1,53 @@
...
+# TODO the system is still using the `non-ac` versions of the files yet.
 
-Release Date: 2017-03-08
-Version: 1
-Publisher: DISA
-Source: STIG.DOD.MIL
-uri: http://iase.disa.mil
------------------
-=end
+MIN_REUSE_GENERATIONS = attribute('min_reuse_generations', default: 5,
```
Comment (trevor-vaughan) [inline controls/V-71933.rb]: Globals need to be removed

File: controls/V-71945.rb [hunk 2]
```diff
@@ -1,79 +1,71 @@
...
+  ]
 
   describe pam('/etc/pam.d/password-auth') do
-    required_lines = [
-      'auth required pam_faillock.so even_deny_root',
-      'auth sufficient pam_unix.so try_first_pass',
-      'auth [default=die] pam_faillock.so even_deny_root'
-    ]
+    its('lines') { should match_pam_rules(required_lines) }
+    its('lines') { should match_pam_rule('auth .* pam_faillock.so (preauth|authfail)').all_with_args('even_deny_root') }
```
Comment (trevor-vaughan) [inline controls/V-71945.rb]: You can modify the `required_lines` above to include the regex so that you don't need multiple tests.

Comment (samcornwell) [reply controls/V-71945.rb]: @trevor-vaughan Part of the reason I included this pattern is to say that "if you have any other line using this module (using preauth or authfail), it must also use `even_deny_root`"  Do you think that's excessive?

Comment (trevor-vaughan) [reply controls/V-71945.rb]: @samcornwell It doesn't really hurt anything but there may be reasons that someone has added rules without that (I can't think of any but PAM can be fun).

Comment (samcornwell) [reply controls/V-71945.rb]: @trevor-vaughan So I can leave it? If so this is good to go, I've gone ahead and removed the constant attributes.

Comment (trevor-vaughan) [reply controls/V-71945.rb]: @samcornwell Yeah, we can leave it.

**Pull request merged**